driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -220,15 +222,11 @@ public Session getSession() {
     private <T> Mapper<T> getMapper(Class<T> klass) {
         Mapper<T> mapper = (Mapper<T>) mappers.get(klass);
         if (mapper == null) {
-            synchronized (mappers) {
-                mapper = (Mapper<T>) mappers.get(klass);
-                if (mapper == null) {
-                    EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, this);
-                    mapper = new Mapper<T>(this, klass, entityMapper);
-                    Map<Class<?>, Mapper<?>> newMappers = new HashMap<Class<?>, Mapper<?>>(mappers);
-                    newMappers.put(klass, mapper);
-                    mappers = newMappers;
-                }
+            EntityMapper<T> entityMapper = AnnotationParser.parseEntity(klass, this);
+            mapper = new Mapper<T>(this, klass, entityMapper);
+            Mapper<T> old = (Mapper<T>) mappers.putIfAbsent(klass, mapper);
+            if (old != null) {
+                mapper = old;
```

Comment:
🐼  my bad, I was looking at the diff view on github and I only focused on the changed lines.